static/js/event-display.js

```diff
@@ -134,6 +82,69 @@ function loadEvents() {
   });
 }
 
+function parseEvent(eventData) {
+  const event = {
+    // TODO: validity tests ?
+
+    roomFloor: eventData.resource.floor,
+    roomName: eventData.resource.title,
+
+    title: eventData.title,
+    subtitle: eventData.subtitle,
+
+    // Convert dates
+    start: moment(eventData.start),
+    end: moment(eventData.end),
+  };
+
+  // Adjust timezone
+  if (typeof tzdelta !== 'undefined' && tzdelta) {
+    event.start = event.start.add(tzdelta, 'hour');
+    event.end = event.end.add(tzdelta, 'hour');
+
+    // Detect Daylight Savings Time
+  } else if (event.start.isDST()) {
+    event.start = event.start.add(1, 'hour');
+    event.end = event.end.add(1, 'hour');
+  }
+
+  return event;
+}
+
+function eventsTable(events) {
+  let html = ''
+  html += '<table class="table room-table"><tbody>';
+  html +=   events.map(eventRow).join();
+  html += '</tbody></table>';
+  return html;
+}
+
+function eventRow(event) {
+  let html = '';
+
+  const rowClasses = []
+  if (event.end.isBefore()) rowClasses.push('event-ended');
+  if (isBrownbag(event)) rowClasses.push('brownbag');
+  html += '<tr class="' + rowClasses.join(' ') + '">';
+  html +=   '<td class="event-time">';
+  html +=     event.start.format('HH:mm') + ' &ndash; ' + event.end.format('HH:mm');
+  html +=   '</td>';
+  html +=   '<td>';
+  html +=     '<div class="event-title">' + event.title + '</div>';
+  html +=     '<div class="event-subtitle">' + event.subtitle + '</div>';
+  html +=   '</td>';
+  html +=   '<td>';
+  html +=     '<div class="room-floor">' + event.roomFloor + '</div>';
+  html +=     '<div class="room-name">' + event.roomName + '</div>';
+  html +=   '</td>';
+  html += '</tr>';
+  return html;
+}
+
+function isBrownbag(event) {
```

Comment:
I'd propose that you deploy this if you are at Effinger only and then restart the beamer, to see if it still works. Some trivial stuff broke with my last brownbag change on the raspberry pi which worked fine on my browser.

Reply:
Was it visual stuff or fetching the API data that went wrong?